src/twisted/names/test/test_names.py

```diff
@@ -1221,13 +1223,50 @@ def loadBindString(self, s):
         @return: a new bind authority
         @rtype: L{twisted.names.authority.BindAuthority}
         """
-        fp = FilePath(self.mktemp().encode("ascii"))
+        fp: FilePath[AnyStr] = FilePath(path)
         fp.setContent(s)
 
         return authority.BindAuthority(fp.path)
 
     def setUp(self):
-        self.auth = self.loadBindString(sampleBindZone)
+        path = self.mktemp().encode("ascii")
+        self.auth = self.loadBindString(sampleBindZone, path)
+
+    def test_loadBindZonePathAsString(self):
+        """
+        L{BindAuthority} loads a BIND zone with filepath as type string.
+        """
+        path = self.mktemp()
+        authority = self.loadBindString(sampleBindZone, path)
+
+        self.assertIsInstance(
+            authority, BindAuthority, "Loaded object is not a BindAuthority"
+        )
+
+        self.assertTrue(authority.records, "No records were loaded from the BIND zone")
+
+        self.assertIsInstance(authority.records, dict, "Records is not a dictionary")
+        self.assertGreater(
+            len(authority.records), 0, "No domains were parsed into records"
+        )
+
+    def test_loadBindZonePathAsBytes(self):
```

Comment:
New tests should have type annotations so they can be checked.
```suggestion
    def test_loadBindZonePathAsBytes(self) -> None:
```